src/Utilities.js

```diff
@@ -76,3 +76,27 @@ function extend_(destination, source) {
   }
   return destination;
 }
+
+/* exported getValueCaseInsensitive_ */
+/**
+ * Gets the value stored in the object under the given key, in a
+ * case-insensitive way.
+ * @param {Object} obj The object to search in.
+ * @param {string} key The key to search for.
+ * @return {Object} the value under that key, or undefined otherwise
+ */
+function getValueCaseInsensitive_(obj, key) {
+  if (obj == null || typeof obj !== 'object' ||
+      key == null || !key.toString) {
+    return undefined;
```

Comment:
Undefined is returned by default in JavaScript.
Why don't you just `return;`?

Sample:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/return#Description